typing/includemod_errorprinter.ml

```diff
@@ -173,18 +180,59 @@ module Illegal_permutation = struct
       Style.inline_code (Ident.name id)
 
   let pp ctx_printer env ppf (mty,c) =
+    match transposition c with
+    | None | Some (_, (Primitive_coercion _ | Alias_coercion _)) ->
+        (* those kind coercions are not inversible, and raise an error earlier when
+           checking for module type equivalence *)
+        assert false
+    | Some (path, Transposition (k,l)) ->
     try
-      let p, k, l = transposition c in
-      let ctx, mt = find env p mty in
+      let ctx, mt = find env path mty in
       Format.fprintf ppf
         "@[<hv 2>Illegal permutation of runtime components in a module type.@ \
-         @[For example,@ %a@]@ @[the %a@ and the %a are not in the same order@ \
+         @[For example%a,@]@ @[the %a@ and the %a are not in the same order@ \
          in the expected and actual module types.@]@]"
         ctx_printer ctx pp_item (item mt k) pp_item (item mt l)
     with Not_found -> (* this should not happen *)
       Format.fprintf ppf
         "Illegal permutation of runtime components in a module type."
 
+  let coercion_in_package_subtype ctx_printer env mty c ppf =
+    match transposition c with
+    | None ->
+        (* The coercion looks like the identity but was not simplified to
+           [Tcoerce_none], this only happens when the two first-class module
+           types differ by runtime size *)
+        Format.fprintf ppf
+          "The two first-class module types differ by their runtime size."
+    | Some (path, c) ->
+  try
+    let ctx, mt = find env path mty in
+    match c with
+    | Primitive_coercion prim_name ->
+        Format.fprintf ppf
+          "@[The two first-class module types differ by a coercion of@ \
+           the primitive %a@ to a value%a.@]"
+          Style.inline_code prim_name
+          ctx_printer ctx
+    | Alias_coercion path ->
+        Format.fprintf ppf
+          "@[The two first-class module types differ by a coercion of@ \
+           a module alias %a@ to a module%a.@]"
+          (Style.as_inline_code Printtyp.path) path
+          ctx_printer ctx
+    | Transposition (k,l) ->
+        Format.fprintf ppf
+          "@[@[The two first-class module types do not share@ \
+           the same position for runtime components.@]@ \
+           @[For example,%a@ the %a@ occurs at the expected position of@ \
+           the %a.@]@]"
+          ctx_printer ctx pp_item (item mt l) pp_item (item mt k)
+  with Not_found ->
+    Format.fprintf ppf
+      "@[The two packages types do not share@ \
+       the@ same@ position@ for@ runtime@ components.@]"
```

Comment:
Maybe `positions` (plural) ?

Reply:
I agree that `positions` sounds slightly better(even if this path should not happen).